components/director/internal/tenantfetchersvc/tenant_fetcher_job_config_test.go

```diff
@@ -0,0 +1,208 @@
+				environ = append(environ, varName+"="+name)
+			}
+
+			// WHEN
+			jobNamesFromEnv := GetJobsNames(ReadFromEnvironment(environ))
+
+			// THEN
+			if testCase.ReadSuccess == true {
+				for _, name := range testCase.JobNames {
+					assert.Contains(t, jobNamesFromEnv, name, "Job names should contain: "+name)
+				}
+			} else {
+				for _, name := range testCase.JobNames {
+					assert.NotContains(t, jobNamesFromEnv, name, "Job names not expected to contain: "+name)
+				}
+			}
+		})
+	}
+}
+
+func TestFetcherJobConfig_ReadEventsConfig(t *testing.T) {
+	// GIVEN
+	jobName := "testJob"
+
+	accountRegion := "testRegion"
+	defaultAccountRegion := "central"
+
+	tenantCreatedEndpoint := "create"
+	defaultTenantCreatedEndpoint := ""
+
+	testCases := []struct {
+		Name        string
+		JobName     string
+		EnvVars     map[string]string
+		ReadFailure bool
+	}{
+		{
+			Name:    "Read events configuration from environment",
+			JobName: jobName,
+			EnvVars: map[string]string{"APP_%s_ACCOUNT_REGION": accountRegion, "APP_%s_ENDPOINT_TENANT_CREATED": tenantCreatedEndpoint},
+		},
+		{
+			Name:    "Get default events configurations when no environment variables",
+			JobName: jobName,
+			EnvVars: nil,
+		}, {
+			Name:        "Get default events configurations when environment variables not match",
+			JobName:     jobName,
+			EnvVars:     map[string]string{"APP2_%s_ACCOUNT_REGION": accountRegion, "APP2_%s_ENDPOINT_TENANT_CREATED": tenantCreatedEndpoint},
+			ReadFailure: true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			var environ []string
+			for k, v := range testCase.EnvVars {
+				varName := fmt.Sprintf(k, testCase.JobName)
+				environ = append(environ, varName+"="+v)
+			}
+
+			// WHEN
+			jobConfig := NewTenantFetcherJobEnvironment(context.TODO(), testCase.JobName, ReadFromEnvironment(environ)).ReadJobConfig()
+			eventsCfg := jobConfig.GetEventsCgf()
+
+			// THEN
+			if testCase.EnvVars == nil || testCase.ReadFailure {
+				assert.Equal(t, defaultAccountRegion, eventsCfg.AccountsRegion, fmt.Sprintf("Default account region should be %s", defaultAccountRegion))
+				assert.Equal(t, defaultTenantCreatedEndpoint, eventsCfg.APIConfig.EndpointTenantCreated, fmt.Sprintf("Default tenant created endpint should be %s", defaultTenantCreatedEndpoint))
+			} else {
+				assert.Equal(t, accountRegion, eventsCfg.AccountsRegion, fmt.Sprintf("Account region should be %s", accountRegion))
+				assert.Equal(t, tenantCreatedEndpoint, eventsCfg.APIConfig.EndpointTenantCreated, fmt.Sprintf(" Tenant created endpint should be %s", tenantCreatedEndpoint))
+			}
+		})
+	}
+}
+
+func TestFetcherJobConfig_ReadHandlerConfig(t *testing.T) {
+	// GIVEN
+	jobName := "testJob"
+
+	jobIntervalMins := 3 * time.Minute
+	defaultJobIntervalMins := 1 * time.Minute
+
+	tenantProvider := "testProvider"
+	defaultTenantProvider := "external-provider"
+
+	testCases := []struct {
```

Comment:
we can have "explicit" test cases here instead of the approach with the loop, it will probably look more simple.
The same applies for the test above